framework/src/media/utils/MediaUtils.cpp

```diff
@@ -1266,20 +1302,27 @@ float getSignalToNoiseRatio(const short *buffer, size_t size, int windows, int *
 	return energyRatio;
 }
 
-bool isMpeg2Ts(const unsigned char *buffer, size_t size)
+audio_type_t getAudioTypeFromStream(const unsigned char *buffer, size_t size)
 {
-	if (!buffer) {
-		meddbg("Invalid param, buffer is null!\n");
-		return false;
-	}
+	audio_type_t audioType;
+	unsigned int channel;
+	unsigned int sampleRate;
+	audio_format_type_t pcmFormat;
 
-	int ret = findTsPacket(buffer, size, true);
-	if (ret < UTILS_ERROR_NONE) {
-		meddbg("Not found TS packets, err %d\n", ret);
-		return false;
+	if (isMpeg2Ts(buffer, size)) {
+		audioType = AUDIO_TYPE_MP2T;
+	} else if (isMp3(buffer, size)) {
+		audioType = AUDIO_TYPE_MP3;
+	} else if (buffer_header_parsing(buffer, size, AUDIO_TYPE_AAC, &channel, &sampleRate, &pcmFormat)) {
+		audioType = AUDIO_TYPE_AAC;
+	} else if (buffer_header_parsing(buffer, size, AUDIO_TYPE_WAVE, &channel, &sampleRate, &pcmFormat)) {
```

Comment:
from line 1316 to 1318,   buffer_header_parsing  function is called twice, 
why not return its type for one call, instead of called twice.

Reply:
buffer_header_parsing(,,AUDIO_TYPE_AAC,,,) is trying to parse AAC header,
if succeed, that we can make decision it is AAC ADTS stream.
As well, buffer_header_parsing(,,AUDIO_TYPE_WAVE,,,) check if it is WAV foramt.